storage/cloudfrontlayerhandler.go

```diff
@@ -74,13 +75,28 @@ func newCloudFrontLayerHandler(storageDriver storagedriver.StorageDriver, option
 
 	cf := cloudfront.New(baseURL, privateKey, keypairID)
 
-	return &cloudFrontLayerHandler{cloudfront: cf, delegateLayerHandler: dlh}, nil
+	duration := 20 * time.Minute
+	d, ok := options["duration"]
+	if ok {
+		switch d := d.(type) {
+		case time.Duration:
+			duration = d
+		case string:
+			dur, err := time.ParseDuration(d)
+			if err != nil {
+				return nil, fmt.Errorf("Invalid duration: %s", err)
+			}
+			duration = dur
```

Comment:
We probably want a default error case for types that are unexpected. It will save some time if a bad configuration is encountered.